### ISSUE TITLE
import/export: false positives for typescript type + value export

### ISSUE THREAD

**Issue by bradzacher** [benmosher/eslint-plugin-import#1318]
Reporting from https://github.com/typescript-eslint/typescript-eslint/issues/10
The following code is valid typescript (even though it's a bit weird).

```TS
// unexected error: Multiple exports of name 'Foo'. (import/export)
export const Foo = 1;
// unexected error: Multiple exports of name 'Foo'. (import/export)
export type Foo = string;
```

When importing this, it causes the name to work as both a type and a value.
```TS
import { Foo } from './foo';

const x: Foo = 'string val';
const y = Foo; // === 1
```

**Comment by ljharb** [comment]
This seems like the same issue as #923.

**Comment by bradzacher** [comment]
@ljharb - looking at #923 - that is about a different rule (named).
Also it also looks like it's about flow, where the code in the OP is not valid.
